web/src/localstorage.js

```diff
@@ -32,9 +32,13 @@ const ls = {
 
     getData(version, name) {
         const key = this.formGetter(version, name);
-        let data = localStorage.getItem(key);
-        data = ls.parseJSON(data);
-
+        let data;
+        try {
+            data = localStorage.getItem(key);
+            data = ls.parseJSON(data);
+        } catch {
+            // data stays undefined
```

Comment:
I think if the JSON parsing is the step that fails, this won't stay undefined. We should probably fix that by just renaming the first variable to `data_json` or something since the variable reuse is messy anyway, but I also wonder if we want to surface a JSON parsing failure with a `blueslip` error, since invalid JSON there is possibly reflects a bug in our application code.

Reply:
If you look above, `ls.parseJSON` has its own `try`/`catch` to return undefined if it fails.  Which is probably more confusing than inlining that code here.